src/Xamarin.CodeAnalysis/RequiresSuperAttributeAnalyzer.cs

```diff
@@ -0,0 +1,106 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+//
+// RequiresSuperAttributeAnalyzer.cs:
+//
+// Authors:
+//   Madeline Zhang  <v-madzh@microsoft.com>
+//
+// Copyright 2020 Xamarin Inc.
+
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace RequiresSuperAttribute
+{
+	/// <summary>
+	/// Analyzer that reports diagnostics for all methods with the [RequiresSuper] attribute which are overridden without a base.[methodname]() call in the method body
+	/// </summary>
+	[DiagnosticAnalyzer (LanguageNames.CSharp)]
+	public class RequiresSuperAttributeAnalyzer : DiagnosticAnalyzer
+	{
+		public static readonly string DiagnosticId = "XI0004";
```

Comment:
The existing Android analyzers use XAAblah, maybe this diagnostic ID could be something like XIAblah for iOS analyzer?

`public const string DiagnosticId = "XAA1001";`

Filenames should also be updated to conform to the preexisting code, but that can wait for a later PR 😄